code/lib/instrumenter/src/instrumenter.ts

```diff
@@ -167,6 +169,24 @@ export class Instrumenter {
       else this.initialized = true;
     });
 
+    /**
+     * TODO Question: Should this logic be done somewhere else, so the instrumenter is not tied to the highglighting of elements?
+     * for instance:
+     * - instrumenter has an event that returns an HTML element based on the call
+     * - the addon gets `storyState` by accessing __STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__ instead, and then there's no need for any code in the instrumenter.
+     */
+    this.channel.on('storybook/interactions/highlight_element', (e) => {
+      const storyState = this.getState(e.storyId);
+      const [element] = Array.from(storyState.callRefsByResult.entries()).find(([_key, value]) => {
+        return value.__callId__ === e.selector;
+      });
+
+      Promise.resolve(element).then((el) => {
+        const selector = getCssSelector(el);
+        this.channel.emit(HIGHLIGHT, { elements: [selector], color: '#6c1d5c' });
+      });
+    });
```

Comment:
Indeed this does not belong here. Instead it should be in the interactions addon. We can have it use `__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__`. Chromatic does that too.

@JonathanKolnik may have some thoughts about it too.

Reply:
I agree!